test/test_binary_ufuncs.py

```diff
@@ -1359,100 +1359,109 @@ def test_rdiv(self, device, dtype):
+                   (torch.remainder, lambda x, mod: x.remainder_(mod))]
+        for fns in fn_list:
+            # check floating-point tensor fmod/remainder to zero is nan on both CPU and GPU
+            x = make_tensor((10, 10), device=device, dtype=dtype, low=-9, high=9)
+            zero = torch.zeros_like(x)
+
+            self.assertTrue(torch.all(fns[0](x, 0.0).isnan()))
+            self.assertTrue(torch.all(fns[0](x, zero).isnan()))
+            # out
+            out = torch.empty(0, device=device, dtype=dtype)
+            fns[0](x, zero, out=out)
+            self.assertEqual(out.size(), torch.Size([10, 10]))
+            self.assertTrue(torch.all(out.isnan()))
+            # in-place
+            fns[1](x, zero)
+            self.assertTrue(torch.all(x.isnan()))
 
     @onlyOnCPUAndCUDA  # Check Issue https://github.com/pytorch/pytorch/issues/48130
     @skipCUDAIfRocm  # Error happens on both ROCM and XLA
     @dtypes(*torch.testing.get_all_int_dtypes())
-    def test_fmod_by_zero_integral(self, device, dtype):
-        # check integral tensor fmod to zero
-        x = make_tensor((10, 10), device=device, dtype=dtype, low=-9, high=9)
-        zero = torch.zeros_like(x)
-        # out
-        out = torch.empty(0, device=device, dtype=dtype)
-        # In-place
-        x_ = x.clone()
-        # RuntimeError on CPU
-        if device == 'cpu':
-            with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
-                x.fmod(zero)
-            with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
-                torch.fmod(x, zero, out=out)
-            with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
-                x.fmod_(zero)
-        # Different value for different dtype on GPU
-        else:
-            if dtype == torch.int64:
-                self.assertEqual(x.fmod(zero) == 4294967295, x >= 0)
-                self.assertEqual(x.fmod(zero) == -1, x < 0)
-                # out
-                torch.fmod(x, zero, out=out)
-                self.assertEqual(out == 4294967295, x >= 0)
-                self.assertEqual(out == -1, x < 0)
-                self.assertEqual(out.size(), torch.Size([10, 10]))
-                # in-place
-                x_.fmod_(zero)
-                self.assertEqual(x_ == 4294967295, x >= 0)
-                self.assertEqual(x_ == -1, x < 0)
+    def test_fmod_remainder_by_zero_integral(self, device, dtype):
+        fn_list = [(torch.fmod, lambda x, mod: x.fmod_(mod)),
+                   (torch.remainder, lambda x, mod: x.remainder_(mod))]
+        for fns in fn_list:
+            # check integral tensor fmod/remainder to zero
+            x = make_tensor((10, 10), device=device, dtype=dtype, low=-9, high=9)
+            zero = torch.zeros_like(x)
+            # out
+            out = torch.empty(0, device=device, dtype=dtype)
+            # In-place
+            x_ = x.clone()
+            # RuntimeError on CPU
+            if device == 'cpu':
```

Comment:
`if self.device_type == 'cpu'` or `if torch.device(device).type == 'cpu'`

Reply:
Since so much of PyTorch's code is copy-pasted it's important to use the blessed patterns, even when another, possibly simpler pattern, will work. For example, if a test tried to do `if device == 'cuda':` that would fail, because device could be `cuda:0` or `cuda:1` when the test is run on a CUDA device. That's why it's important to check the device's type using `self.device_type` or constructing a device with `torch.device(device)` and then acquiring its type.

Reply:
It makes a lot of sense!